### PR TITLE
ISSUE-165: Provide Twig variables for Ajax Preview in Twig templates

### LINKED ISSUE #165: Provide Twig variables for Ajax Preview in Twig templates
# What?
Corresponding issue for esmero/format_strawberryfield#274. The same functionality needs to be provided here when previewing AMI sets against Twig templates for ingest.

### PR DESCRIPTION
**Pull request by aksm**
Resolves #165. Requires esmero/format_strawberryfield#306 to be merged first.

### CODE REVIEW

**Comment by DiegoPino** [comment]
@aksm also, please add the labels and Milestone? (0.6.0)

File: ami.module [hunk 1]
```diff
@@ -128,6 +131,19 @@ function ami_form_metadatadisplay_entity_edit_form_alter(&$form,FormStateInterfa
...
+  $form['preview']['show_json_table'] = [
+    '#type' => 'checkbox',
+    '#defaut_value' => FALSE,
+    '#title' => 'Show Preview with JSON keys used in this template',
+    '#states' => [
+      'visible' => [
+        [':input[name="ado_context_preview"]' => ['filled' => true]],
+        'or',
+        [':input[name="ado_amiset_preview"]' => ['filled' => true],
+         ':input[name="ado_amiset_row_context_preview"]' => ['filled' => true]]
```
[the remark this answers is not in the document]
Comment (aksm) [reply ami.module]: Hi @DiegoPino, I at least cleaned up the lines for readability: e71442d

Does that make more sense, or am I still missing something?

File: src/Controller/AmiRowAutocompleteHandler.php [hunk 2]
```diff
@@ -394,6 +423,16 @@ public static function ajaxPreviewAmiSet($form, FormStateInterface $form_state)
                   '#markup' => $render,
                 ],
               ];
+              if($show_json_table) {
```
[the remark this answers is not in the document]
Comment (aksm) [reply src/Controller/AmiRowAutocompleteHandler.php]: @DiegoPino Thanks for catching that. I refactored this and everything else to match 1:1: 9483614

Incidentally, I forgot to add the codemirror change to format_strawberryfield so will open an issue and update there.

File: ami.module (hunk 1 above)
[the remark this answers is not in the document]
Comment (DiegoPino) [reply ami.module]: Oh that makes sense. Seeing arrays in diffs is hard. thx

File: ami.module [hunk 3]
```diff
@@ -33,7 +33,10 @@ function ami_help($route_name, RouteMatchInterface $route_match) {
   }
 }
 
-function ami_form_metadatadisplay_entity_edit_form_alter(&$form,FormStateInterface $form_state, $form_id) {
+/**
+ * Implements hook_form_alter().
+ */
+function ami_form_metadatadisplay_entity_form_alter(&$form,FormStateInterface $form_state, $form_id) {
```
Comment (DiegoPino) [inline ami.module:39]: @aksm ok, so I think I asked this before. But this will not add a preview option to the "delete" ami set right?

Comment (aksm) [reply ami.module:39]: @DiegoPino correct, no preview option added to delete.

**Review by DiegoPino — APPROVED** [review]

**Comment by DiegoPino** [comment]
Ok, will merge and test again tomorrow with real data/full 1.2.0 ensemble. thx

**Pull request merged**